### PR TITLE
Matrix fix for KBD75 round 1&2

### PR DESCRIPTION
**Pull request by gazer000**
## Description

- KBD75 round 1&2, even though using ps2avrGB (same as B.mini), appears to have a different key matrix, upon flashing B.mini hex onto these boards, some keys are incorrectly mapped while 1-2 keys may not work

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mechmerlin** [comment]
I would prefer that the directory naming would be

kbdfans/kbd75/bmc or kbdfans/kbd75/ps2avrGB

also, don't forget to update the readme located in the kbdfans/kbd75 directory to point to these new ones.

File: keyboards/kbdfans/kbd75round1/kbd75round1.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  { K00, K10,   K20, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KB0, KC0,   KD0,   KE0   }, \
+  { K01, K11,   K21, K31,   K41,   K51,   KC_NO, KC_NO, KC_NO, KC_NO, KA1,   KB1, KC_NO, KD1,   KE1   }, \
+  { K02, K12,   K22, K32,   K42,   K52,   KC_NO, KC_NO, KC_NO, KC_NO, KA2,   KB2, KC_NO, KD2,   KE2   }, \
+  { K03, K13,   K23, K33,   K43,   K53,   KC_NO, KC_NO, KC_NO, KC_NO, KA3,   KB3, KC3,   KD3,   KC_NO }, \
+  { K04, K14,   K24, K34,   K44,   K54,   KC_NO, KC_NO, KC_NO, KC_NO, KA4,   KB4, KC4,   KD4,   KE4   }, \
+  { K05, KC_NO, K25, K35,   K45,   K55,   KC_NO, KC_NO, KC_NO, KC_NO, KC_NO, KB5, KC5,   KD5,   KE5   }, \
+  { K06, K16,   K26, K36,   K46,   K56,   K66,   K76,   K86,   K96,   KA6,   KB6, KC6,   KD6,   KE6   }, \
+  { K07, K17,   K27, K37,   K47,   K57,   K67,   K77,   KC_NO, KC_NO, KA7,   KB7, KC7,   KD7,   KE7   } \
+}
+
```
Comment (spidey3) [inline keyboards/kbdfans/kbd75round1/kbd75round1.h:56]: Can you apply your fix to the LAYOUT_splitbs too?  The only difference should be split backspace support.

File: keyboards/kbdfans/kbd75round1/readme.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# KBD75 round 1 & 2
+
+QMK Firmware is not the official firmware for the KBD75 round 1 & 2. 
+
+* Keyboard Maintainer: Anh Nguyen
+* Hardware Supported: KBD75 round 1 & 2 PCB
+
```
Comment (spidey3) [inline keyboards/kbdfans/kbd75round1/readme.md:7]: Please add a link to a pcb image, and some text describing how to discern this vs. other "kbd75" variants.

**Review by spidey3 — CHANGES_REQUESTED** [review]
Please update LAYOUT_splitbs too.

**Comment by gazer000** [comment]
> I would prefer that the directory naming would be
> 
> kbdfans/kbd75/bmc or kbdfans/kbd75/ps2avrGB
> 
> also, don't forget to update the readme located in the kbdfans/kbd75 directory to point to these new ones.
> 
> Also, there is a chance that there are other discrepencies here, as I've build a few BMC KBD75s, and the b mini hex file worked just fine.

My PCB is a white v1 PCB (probably round 2) that looks exactly like any other v1 PCB but without the reset button and uses atmega32a instead of 32u4. I'll rename the folder to my exact PCB (which is a round 2 PCB) instead of generalizing this for both round 1 and 2 PCBs.

File: keyboards/kbdfans/kbd75_round2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = bootloadHID
+
+# build options
+BOOTMAGIC_ENABLE = full
```
Comment (spidey3) [inline keyboards/kbdfans/kbd75_round2/rules.mk:15]: We prefer [Bootmagic Lite](https://docs.qmk.fm/#/feature_bootmagic?id=bootmagic-lite) for any new keyboards, as it eliminates a major source of confusion around key swaps, disabled keys, eeprom reset, etc.
```suggestion
BOOTMAGIC_ENABLE = lite
```

**Review by spidey3 — CHANGES_REQUESTED** [review]
Sorry - one more small change requested.

**Review by spidey3 — APPROVED** [review]

**Review by zvecr — CHANGES_REQUESTED** [review]
> I would prefer that the directory naming would be
> 
> kbdfans/kbd75/bmc or kbdfans/kbd75/ps2avrGB

I 2nd this, and would rather see it under the existing kbd75 folder as a sub revision. As for revision name, `round2` would be fine with me. (if going with "ps2avrGB", it would have to be lower case)

**Comment by spidey3** [comment]
@gazer000 can you please check whether `ymd75/rev1` firmware works on your keyboard as you would expect, with the correct key mappings? We have some evidence that this might be the case.

Also, please verify the mappings of the Del vs. Backspace for the splitbs LAYOUT.  I suspect they might be reversed.

**Comment by gazer000** [comment]
> @gazer000 can you please check whether `ymd75/rev1` firmware works on your keyboard as you would expect, with the correct key mappings? We have some evidence that this might be the case.
> 
> Also, please verify the mappings of the Del vs. Backspace for the splitbs LAYOUT. I suspect they might be reversed.

It seems that what you're suspecting is true, the ymd75/rev1 firmware does work on my board with the correct key mappings

**Comment by spidey3** [comment]
> It seems that what you're suspecting is true, the ymd75/rev1 firmware does work on my board with the correct key mappings - the header files look identical

If that firmware works on your keyboard, I think I would like to close this PR, and file a new one to update `kbd75/readme.md` to indicate that round2 == ymd75/rev1.

**Pull request closed without merging**